src/components/NeedsView.tsx

```diff
@@ -1,26 +1,224 @@
 import React, { Component } from 'react';
-import { View, Text, StyleSheet } from 'react-native';
+import {
+    View,
+    Text,
+    FlatList,
+    TouchableOpacity,
+    Dimensions,
+    KeyboardAvoidingView,
+    Modal
+} from 'react-native';
+import MapView from 'react-native-maps'
+import { TextCell } from './TextCell'
+import { ButtonCell } from './ButtonCell'
+import { CategoryList } from './CategoryList'
+import { API } from './../API/API'
+import { Colors } from './Colors'
 
-export class NeedsView extends Component<{}, {}> {
+export enum MarkerValue {
+    Name,
+    Phone,
+    Address,
+    Description,
+    Email
+}
+interface State {
+    currentLocation: {
+        latitude: number,
+        longitude: number
+    },
+    modalVisible: boolean,
+    selectedCategories: {},
+    phone: string,
+    name: string,
+    address: string
+    email?: string,
+    description: string
+
+}
+export class NeedsView extends Component<{}, State> {
+
+    constructor(props) {
+        super(props)
+        this.state = {
+            currentLocation: null,
+            modalVisible: false,
+            selectedCategories: [],
+            phone: '',
+            description: '',
+            email: '',
+            address: '',
+            name: ''
+        }
+    }
+
+    renderItem = ({ item, index }: { item: string, index: number }) => {
```

Comment:
Why are some of the methods declared as closures assigned to properties? Is there an advantage to this pattern that I just don't know, yet?